ChelseasFieldJournal/src/GUI/GuiComposerHelpers.cs

```diff
@@ -0,0 +1,836 @@
+        }
+
+        public static GuiComposer AddToggleButton(this GuiComposer composer, string text, CairoFont font, Action<bool> onToggle, ElementBounds bounds, string key = null)
+        {
+            if (!composer.Composed)
+            {
+                composer.AddInteractiveElement(new GuiElementToggleButton(composer.Api, "", text, font, onToggle, bounds, true), key);
+            }
+            return composer;
+        }
+
+        public static GuiComposer AddIconButton(this GuiComposer composer, string icon, Action<bool> onToggle, ElementBounds bounds, string key = null)
+        {
+            if (!composer.Composed)
+            {
+                composer.AddInteractiveElement(new GuiElementToggleButton(composer.Api, icon, "", CairoFont.WhiteDetailText(), onToggle, bounds, false), key);
+            }
+            return composer;
+        }
+        public static void ToggleButtonsSetValue(this GuiComposer composer, string key, int selectedIndex)
+        {
+            int i = 0;
+            GuiElementToggleButton btn;
+            while ((btn = composer.GetToggleButton(key + "-" + i.ToString())) != null)
+            {
+                btn.SetValue(i == selectedIndex);
+                i++;
+            }
+        }
+        public static GuiComposer AddIconToggleButtons(this GuiComposer composer, string[] icons, CairoFont font, Action<int> onToggle, ElementBounds[] bounds, string key = null)
+        {
+            if (!composer.Composed)
+            {
+                int quantityButtons = icons.Length;
+                for (int i = 0; i < icons.Length; i++)
+                {
+                    int index = i;
+                    composer.AddInteractiveElement(new GuiElementToggleButton(composer.Api, icons[i], "", font, delegate (bool on)
+                    {
+                        if (on)
+                        {
+                            onToggle(index);
+                            for (int j = 0; j < quantityButtons; j++)
+                            {
+                                if (j != index)
+                                {
+                                    composer.GetToggleButton(key + "-" + j.ToString()).SetValue(false);
+                                }
+                            }
+                            return;
+                        }
+                        composer.GetToggleButton(key + "-" + index.ToString()).SetValue(true);
+                    }, bounds[i], true), key + "-" + i.ToString());
+                }
+            }
+            return composer;
+        }
+
+        public static GuiComposer AddTextToggleButtons(this GuiComposer composer, string[] texts, CairoFont font, Action<int> onToggle, ElementBounds[] bounds, string key = null)
+        {
+            if (!composer.Composed)
+            {
+                int quantityButtons = texts.Length;
+                for (int i = 0; i < texts.Length; i++)
```

Comment:
There must be a better way than nesting i, j for loops. This can get slow with many elements.